databuilder/databuilder/task/search/search_metadata_to_elasticsearch_task.py

```diff
@@ -0,0 +1,140 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Dict, Iterator
+
+from pyhocon import ConfigTree
+
+from elasticsearch_dsl.connections import connections, Connections
+from elasticsearch.helpers import parallel_bulk
+from elasticsearch_dsl.document import Document
+from elasticsearch_dsl.index import Index
+
+from databuilder import Scoped
+from databuilder.task.base_task import Task
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.transformer.base_transformer import NoopTransformer, Transformer
+from databuilder.utils.closer import Closer
+
+from databuilder.task.search.document_mappings import RESOURCE_TO_MAPPING, english_analyzer, stemming_analyzer
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SearchMetadatatoElasticasearchTask(Task):
+    
+    
+    ELASTICSEARCH_CLIENT_CONFIG_KEY = 'client'
+    ENTITY_TYPE = 'doc_type'
+    MAPPING = 'document_mapping'
+    CUSTOM_ANALYZERS = 'custom_analyzers'
+    ELASTICSEARCH_NEW_INDEX_CONFIG_KEY = 'new_index'
+    ELASTICSEARCH_ALIAS_CONFIG_KEY = 'alias'
+
+    # config to control how many max documents to publish at a time
+    ELASTICSEARCH_PUBLISHER_BATCH_SIZE = 'batch_size'
+
+    def __init__(self,
+                 extractor: Extractor,
+                 transformer: Transformer = NoopTransformer()) -> None:
+        self.extractor = extractor
+        self.transformer = transformer
+
+        self._closer = Closer()
+        self._closer.register(self.extractor.close)
+        self._closer.register(self.transformer.close)
+
+    def init(self, conf: ConfigTree) -> None:
+        self.extractor.init(Scoped.get_scoped_conf(conf, self.extractor.get_scope()))
+        self.transformer.init(Scoped.get_scoped_conf(conf, self.transformer.get_scope()))
+
+        conf = Scoped.get_scoped_conf(conf, self.get_scope())
+        default_analyzers = [english_analyzer, stemming_analyzer]
+
+        self.entity = conf.get_string(SearchMetadatatoElasticasearchTask.ENTITY_TYPE, default='table').lower()
+        self.document_mapping = conf.get(SearchMetadatatoElasticasearchTask.MAPPING, RESOURCE_TO_MAPPING[self.entity])
+        # not sure if this is the best way to configure custom analyzers since it is not enough to add them to the documents
+        self.analyzers = conf.get_list(SearchMetadatatoElasticasearchTask.CUSTOM_ANALYZERS, default_analyzers)
```

Comment:
I think the second part of the story would be to have`SearchableResource` accept custom analyzers in the constructor. Simple idea would be to pass it in the form of dict `{'name_of_analyzer': spec_of_analyzer}` and then the analyzer would be appended to every field as a subfield. For example `SearchableResource(analyzers={'polish': polish_analyzer_spec})` would create additional `name.polish`, `description.polish` ... etc. fields

And the third side of the story is if we do that then maybe some logging is required so users don't forget to adjust search query accordingly in search API.

Reply:
So the thing is that if the analyzers are not added to the index itself rather than just the Documents it doesn't create the mapping with the analyzers, so maybe passing in a custom index would be a better strategy?

Reply:
yea I think this might be redundant. If I wanted to have custom mapping with custom analyzers I would just subclass any of SearchableResource descendants. +1 to removing this.

Reply:
I do but the issue is that for whatever reason I am unable to create an index with the analyzers just by using the Documents, I am looking into this issue now